test/controllers/samples_controller_test.rb

```diff
@@ -207,31 +208,41 @@ class SamplesControllerTest < ActionDispatch::IntegrationTest
 
   test 'should return pipeline stage results' do
     post user_session_path, params: @user_params
-    get stage_results_sample_url(@pipeline_run_sample)
+    get stage_results_sample_url(@public_pipeline_run_sample)
     assert_response :success
 
     results = @response.parsed_body["pipeline_stage_results"]
     assert_equal 4, results.length
-    @pipeline_run.pipeline_run_stages.each do |stage|
+    @public_pipeline_run.pipeline_run_stages.each do |stage|
       assert_equal JSON.parse(stage.dag_json), results[stage.name]
     end
   end
 
-  test 'joe cannot see pipeline experimental stage results' do
```

Comment:
Can you add an admin test to verify it can see the experiment stage? 
Note: As discussed, we want to move tests to rspec/factories... You can do this now, or later on a follow up PR.

Reply:
Added experimental stage check for admin in the test above (+ renamed for better clarity)

I'm thinking about moving tests to rspec/factories in a separate PR just so this PR is cleaner/ doesn't have a huge amount of code for new factories everywhere